src/variant.rs

```diff
@@ -178,6 +273,11 @@ impl Variant {
     pub fn get_data_as_bytes(&self) -> Bytes {
         unsafe { from_glib_full(glib_sys::g_variant_get_data_as_bytes(self.to_glib_none().0)) }
     }
+
+    /// Determines the number of children in a container GVariant instance.
+    pub fn n_children(&self) -> usize {
+        unsafe { glib_sys::g_variant_n_children(self.to_glib_none().0) }
```

Comment:
What does this do on non-container variants? C docs say

> It is an error to call this function on any other type of GVariant.

So we should probably check that here?

Reply:
I assume it's a `g_return_val_if_fail()` in C that causes a) 0 to be returned and not showing the caller that things are wrong, and b) generally `g_return_val_if_fail()` and similar guards are leaving everything behind in an inconsistent state.

If it was a proper assertion then a panic would still be better as that can at least be caught at isolation boundaries if needed.